src/api/server.middleware.ts

```diff
@@ -34,86 +42,63 @@ export class ServerMiddleware {
-
-    const schedule = _.cloneDeep(originalSchedule);
-
-    Object.keys(schedule.shifts).map(
-      (shiftName): void => (nameToUuid[shiftName] = uuidv4(shiftName))
-    );
-
-    Object.keys(schedule.shifts).forEach((shiftName) => {
-      schedule.shifts[nameToUuid[shiftName]] = schedule.shifts[shiftName];
-      delete schedule.shifts[shiftName];
-    });
-
-    Object.keys(schedule.employee_info.time).forEach((shiftName) => {
-      schedule.employee_info.time[nameToUuid[shiftName]] = schedule.employee_info.time[shiftName];
-      delete schedule.employee_info.time[shiftName];
-    });
-
-    Object.keys(schedule.employee_info.type).forEach((shiftName) => {
-      schedule.employee_info.type[nameToUuid[shiftName]] = schedule.employee_info.type[shiftName];
-      delete schedule.employee_info.type[shiftName];
-    });
-
-    Object.keys(schedule.employee_info.team).forEach((shiftName) => {
-      schedule.employee_info.team[nameToUuid[shiftName]] = schedule.employee_info.team[shiftName];
-      delete schedule.employee_info.team[shiftName];
-    });
+  public static mapCodeTypeToKind(
+    error: BackendErrorObject
+  ): {
+    [key in keyof AlgorithmError]?: AlgorithmError[key];
+  } {
+    return { ...error, kind: error.code as AlgorithmErrorCode };
+  }
 
-    if (schedule.employee_info.contractType !== undefined) {
-      Object.keys(schedule.employee_info.contractType).forEach((shiftName) => {
-        schedule.employee_info.contractType![
-          nameToUuid[shiftName]
-        ] = schedule.employee_info.contractType![shiftName];
-        delete schedule.employee_info.contractType![shiftName];
+  public static anonymizeSchedule(schedule: ScheduleDataModel): AnonymizeScheduleReturn {
+    const workerNames = getWorkerNames(schedule);
+    const anonymizedSchedule = _.cloneDeep(schedule) as AnonynimizedSchedule;
+    const nameToUUIDMap = {};
+    workerNames.forEach((workerName) => {
+      const workerUID = uuidv4();
+      nameToUUIDMap[workerName] = workerUID;
+      Object.values(SensitiveDataFieldAccessors).forEach((getFieldObjectFunction) => {
+        const objectWithSensitiveData = getFieldObjectFunction(anonymizedSchedule);
+        const workerData = objectWithSensitiveData[workerName];
+        objectWithSensitiveData[workerUID] = workerData;
+        delete objectWithSensitiveData[workerName];
       });
-    }
-
-    return {
-      anonynimizedSchedule: schedule,
-      anonymizationMap: nameToUuid,
-    };
+    });
+    return [anonymizedSchedule, nameToUUIDMap];
   }
 
   public static mapIsWorkingTypeSnakeCase(schedule: ScheduleDataModel): ScheduleDataModel {
     const result = _.cloneDeep(schedule);
     /* eslint-disable @typescript-eslint/no-explicit-any */
     Object.keys(result.shift_types).forEach((shiftCode) => {
+      const isWorkingShift = result.shift_types[shiftCode]?.isWorkingShift;
+      if (_.isNil(isWorkingShift)) {
+        throw Error(`Shift ${shiftCode} is not defined in shift types`);
+      }
```

Comment:
Jeśli taka sytuacja zajdzie - to coś w naszej aplikacji jest nie tak. 
Według mnie dobrze jest iść w stronę takich wyjątków. 
Trochę śmierdzi Javą, ale my obsługujemy takie wyjątki, więc raczej nie powinno być problemu